credentials/java/com/google/auth/Credentials.java

```diff
@@ -26,6 +26,8 @@
    * populated in headers or other context. The operation can block and fail to complete and may do
    * things such as refreshing access tokens.
    *
+   * <p>If a key ends with {@code "-bin"}, its values are base64-encoded binary data.
+   *
    * @throws IOException if there was an error getting up-to-date access.
```

Comment:
The goal is to plumb binary data from `Crendentials` to its consumer.

The idea is that if a key ends with `-bin`, e.g., `euc-proto-bin`, its values are __guaranteed__ (by `Credentials` implementations) to be __valid base64-encoded binary data__, so that the consumer of the values (in my case [ClientAuthInterceptor](https://github.com/grpc/grpc-java/blob/master/auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java)) can use base64 to decode it and get the binary data. Without such guarantee from the `Crendentials` interface, `ClientAuthInterceptor` wouldn't have a way to plumb binary data from `Crendentials`. It's a contract between the implementations and the users of this method. I can't think of a better place for this to be.